File: back-end/go/internal/mongodb/user_delete.go
package mongodb

import (
	"context"
	"fmt"
	"go.mongodb.org/mongo-driver/bson"
	"go.mongodb.org/mongo-driver/bson/primitive"
	"go.mongodb.org/mongo-driver/mongo"
)

func (d *Datastore) DeleteUser(id string) (int64, error) {
	var objectID primitive.ObjectID
	var err error
	var result *mongo.DeleteResult

	// Parsing a string id to MongoDB ObjectID.
	objectID, err = primitive.ObjectIDFromHex(id)

	if err != nil {
		return 0, fmt.Errorf("the id isn't valid")
	}

	result, err = d.DB.Collection("users").DeleteOne(
		context.Background(),
		bson.M{"_id": objectID},
	)

	if err != nil {
		return 0, err
	}

	return result.DeletedCount, nil
}

File: back-end/nodejs/internal/mongodb/user_delete.js
var User = require('../models/user.js');

async function DeleteUser(id, callback) {
    await User.deleteOne({_id: id})
        .exec()
        .then((doc) => {
            callback(null, doc)
        })
        .catch(err => {
            callback(err, null);
        });
}

exports.DeleteUser = DeleteUser;